Platon/backend/app/workspace_system/views.py

```diff
@@ -0,0 +1,154 @@
+from flask import make_response,jsonify,request
+from flask_restplus import Resource,Namespace, fields
+
+from app.auth_system.helpers import login_required
+from app.workspace_system.forms import TrendingProjectsForm,trending_project_parser,get_self_parser
+from app.workspace_system.models import Workspace,Contribution
+from app.auth_system.models import User
+from app.workspace_system.helpers import schedule_regularly
+
+
+from app import api, db
+
+workspace_system_ns = Namespace("Workspace System",
+                                description="Workspace System Endpoints",
+                                path = "/worksapce_system")
+
+contributor_model = api.model("Contributor", {
+    "id": fields.Integer,
+    "name": fields.String,
+    "surname": fields.String,
+    }
+)
+
+workspace_model = api.model('Worksapce', {
+    "id": fields.Integer,
+    "title": fields.String,
+    "is_private": fields.Integer,
+    "description": fields.String,
+    "state": fields.Integer,
+    "deadline": fields.String,
+    "creation_time": fields.String,
+    "max_contibutors": fields.Integer,
+    "contributors": fields.List(
+        fields.Nested(contributor_model)
+    ),
+})
+
+workspace_list_model = api.model('Worksapce List', {
+    'workspaces': fields.List(
+        fields.Nested(workspace_model)
+    )
+})
+
+workspace_small_model = api.model('Workspace', {
+    "id": fields.Integer,
+    "title": fields.String,
+    "description": fields.String,
+    "state": fields.Integer,
+    "contributors": fields.List(
+        fields.Nested(contributor_model)
+    ),
+})
+
+trending_workspaces_model = api.model('Trending Projects', {
+    'trending_projects': fields.List(
+        fields.Nested(workspace_small_model)
+    )
+})
+
+@workspace_system_ns.route("/self")
+class GetSelfWorkspaces(Resource):
+
+    @api.doc(responses={401:'Authantication Problem',500:'Database Connection Problem'})
+    @api.response(200,'Valid Response',workspace_list_model)
+    @api.expect(get_self_parser)
+    @login_required
+    def get(user_id,self):
+        try:
+            contributions = Contribution.query.filter(Contribution.user_id==user_id).all()
```

Comment:
For this part, I have a short suggestion: instead of getting all the workspaces user contributed to and later filtering them with the for loop and the if statement below, you might change the query by adding "is_active=True" and will not need those.